tests/components/swiss_public_transport/test_config_flow.py

```diff
@@ -0,0 +1,220 @@
+from homeassistant.components.swiss_public_transport.const import (
+    CONF_DESTINATION,
+    CONF_START,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_flow_user_init_data_success(hass: HomeAssistant) -> None:
+    """Test success response."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["handler"] == "swiss_public_transport"
+    assert result["data_schema"] == config_flow.DATA_SCHEMA
+
+    with patch(
+        "homeassistant.components.swiss_public_transport.config_flow.OpendataTransport.async_get_data",
+        autospec=True,
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN, context={"source": "user"}
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_START: "test_start",
+                CONF_DESTINATION: "test_destination",
+            },
+        )
+
+        assert result["type"] == "create_entry"
+        assert result["result"].title == "test_start test_destination"
+
+        assert result["data"] == {
+            CONF_START: "test_start",
+            CONF_DESTINATION: "test_destination",
+        }
+
+
+@pytest.mark.parametrize(
+    ("raise_error", "text_error"),
+    [
+        (OpendataTransportConnectionError(), "cannot_connect"),
+        (OpendataTransportError(), "bad_config"),
+        (IndexError(), "unknown"),
+    ],
+)
+async def test_flow_user_init_data_cannot_connect_error(
+    hass: HomeAssistant, raise_error, text_error
+) -> None:
+    """Test cannot_connect errors."""
+    with patch(
+        "homeassistant.components.swiss_public_transport.config_flow.OpendataTransport.async_get_data",
+        autospec=True,
+        side_effect=raise_error,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN, context={"source": "user"}
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_START: "test_start",
+                CONF_DESTINATION: "test_destination",
+            },
+        )
+
+        assert result["type"] == "form"
+        assert result["errors"]["base"] == text_error
+
+
+@pytest.mark.parametrize(
+    ("raise_error", "text_error"),
+    [
+        (OpendataTransportConnectionError(), "cannot_connect"),
+        (OpendataTransportError(), "bad_config"),
+        (IndexError(), "unknown"),
+    ],
+)
+async def test_flow_user_init_data_unknown_error_and_recover(
```

Comment:
Seems same as the test above but also recovers which is good?

Reply:
It was asked by @joostlek to show that the occurrence of any type of error does not hinder the completion of the config flow.

Reply:
That's correct. I meant you could delete the test above as it's testing same thing as this one (which also recovers)